File: 0007-reverse-integer/0007-reverse-integer.java
class Solution {
    public int reverse(int x) {
        boolean flag = false;
        if (x<0){
            x = Math.abs(x);
            flag = true;
        }

        String str = new StringBuilder(Integer.toString(x)).reverse().toString();
        int reverse = 0;
        try{
            reverse = Integer.parseInt(str);
        }catch(Exception e){
            // do nothing;
        }
        if (flag == true) return (-1)*reverse;
        return reverse;
    }
}

File: 0007-reverse-integer/0007-reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        if x>=0:
            ans = int("".join(reversed(str(x))))
        else:
            ans = int("".join(reversed(str(x)[1:]))) * -1
        if pow(-2,31) <= ans and ans <= pow(2,31)-1:
            return ans
        return 0